airbyte-webapp/src/components/connection/ConnectionForm/DestinationStreamPrefixName.tsx

```diff
@@ -0,0 +1,82 @@
+import { Field, FieldProps, useFormikContext } from "formik";
+import { FormattedMessage, useIntl } from "react-intl";
+
+import { Button } from "components/ui/Button";
+import { FlexContainer } from "components/ui/Flex";
+import { Text } from "components/ui/Text";
+
+import { useConnectionFormService } from "hooks/services/ConnectionForm/ConnectionFormService";
+import { useModalService } from "hooks/services/Modal";
+
+import { FormikConnectionFormValues } from "./formConfig";
+import { FormFieldWrapper } from "./FormFieldWrapper";
+import { ControlLabels } from "../../LabeledControl";
+import {
+  DestinationStreamNamesFormValueType,
+  DestinationStreamNamesModal,
+  StreamNameDefinitionValueType,
+} from "../DestinationStreamNamesModal/DestinationStreamNamesModal";
+
+export const DestinationStreamPrefixName = () => {
+  const { mode } = useConnectionFormService();
+  const { formatMessage } = useIntl();
+  const { openModal, closeModal } = useModalService();
+  const formikProps = useFormikContext<FormikConnectionFormValues>();
+
+  const destinationStreamNamesChange = (value: DestinationStreamNamesFormValueType) => {
+    formikProps.setFieldValue(
+      "prefix",
+      value.streamNameDefinition === StreamNameDefinitionValueType.Prefix ? value.prefix : ""
+    );
+  };
+
+  const destinationModalContent = (
+    <DestinationStreamNamesModal
+      initialValues={{
+        prefix: formikProps.values.prefix,
+      }}
+      onCloseModal={closeModal}
+      onSubmit={destinationStreamNamesChange}
+    />
+  );
```

Comment:
Kinda feel like this should just be in-lined below. Open to discussion.

Reply:
Or memoized since it's referenced by a function.

Reply:
Initially, it was in-lined, but I wanted to make out template flatter.
What about moving the whole onClick openModal handler outside and wrapping it with useCalback? WDYT?

Reply:
That sounds good to me as well.